osgar/drivers/tankmaria.py

```diff
@@ -0,0 +1,149 @@
+"""
+  Driver for a small tank Maria
+"""
+import math
+
+from osgar.node import Node
+from osgar.bus import BusShutdownException
+
+
+ENC_SCALE = 0.00052
+WHEEL_DISTANCE = 0.4  # 50cm long, 40cm wide
+
+
+class MariaTimer(Node):
+    def __init__(self, config, bus):
+        super().__init__(config, bus)
+        bus.register('tick')
+        self.sleep_time = config['sleep']
+
+    def run(self):
+        try:
+            while self.is_alive():
+                self.publish('tick', None)
+                self.sleep(self.sleep_time)
+
+        except BusShutdownException:
+            pass
+
+
+class RobotTankMaria(Node):
+    def __init__(self, config, bus):
+        super().__init__(config, bus)
+        bus.register('pose2d', 'emergency_stop', 'encoders', 'raw')
+
+        # commands
+        self.desired_speed = 0.0  # m/s
+        self.desired_angular_speed = 0.0
+
+        # status
+        self.emergency_stop = None  # uknown state
+        self.pose = (0.0, 0.0, 0.0)  # x, y in meters, heading in radians (not corrected to 2PI)
+        self.buttons = None
+
+        self.verbose = False  # should be in Node
+        self.buf = b''
+        self.encoder = None
+        self.led = 1
+        self.milisec = None
+
+    def send_pose(self):
+        x, y, heading = self.pose
+        self.publish('pose2d', [round(x*1000), round(y*1000),
+                                round(math.degrees(heading)*100)])
+
+
+    def compute_pose(self, left, right):
+        """Update internal pose with 'dt' step"""
+        if left is None or right is None:
+            return False, None, None
+        x, y, heading = self.pose
+
+        metricL = ENC_SCALE * left  # dt is already part of ENC_SCALE
```

Comment:
What is `dt`?

Reply:
delta time

Reply:
how can a time duration be part of encoder scale?

Reply:
sorry - bad copy & paste :( - removed